lua/mapvote/client/vgui/mapvote_panel.lua

```diff
@@ -0,0 +1,261 @@
+        return x + avatarTotalSize * (maxColumnCount - 1), y + (maxRowCount - 1) * avatarTotalSize, false
+    end
+    return x + avatarTotalSize * (mapData.voterCount % maxColumnCount), y + nextRowNumber * avatarTotalSize, true
+end
+
+function PANEL:setup()
+    for _, row in pairs( self.rows ) do
+        row:Remove()
+    end
+    self.rows = {}
+    local count = #self.maps
+    local margin = 2
+    local iconWidth, iconHeight = maxIconSize( self:GetWide(), self:GetTall(), count, 1 )
+    iconWidth = iconWidth - margin * 2
+
+    local maxItemsPerRow = math.floor( self:GetWide() / iconWidth )
+
+    local requiredRows = math.ceil( count / maxItemsPerRow )
+    for rowNumber = 1, requiredRows do
+        local itemsLeft = count - (rowNumber - 1) * maxItemsPerRow
+        local itemsInRow = math.min( maxItemsPerRow, itemsLeft )
+        local rowWidth = (iconWidth + margin * 2) * itemsInRow
+
+        local row = vgui.Create( "Panel", self )
+        table.insert( self.rows, row )
+        row:SetSize( rowWidth, iconHeight )
+        local extraSpace = math.max( 0, self:GetWide() - rowWidth )
+
+        row:SetPos( extraSpace / 2, iconHeight * (rowNumber - 1) )
+        for i = 1, itemsInRow do
+            local index = (rowNumber - 1) * maxItemsPerRow + i
+            local map = self.maps[index]
+            local mapIcon = vgui.Create( "MapVote_MapIcon", row ) --[[@as MapIcon]]
+            ---@diagnostic disable-next-line: duplicate-set-field
+            mapIcon.DoClick = function()
+                self:OnMapClicked( index, map )
+            end
+            mapIcon:SetMap( map.map )
+            mapIcon:SetSize( iconWidth, iconHeight )
+            mapIcon:Dock( LEFT )
+            mapIcon:DockMargin( margin, margin, margin, margin )
+            mapIcon.row = row
+            mapIcon.voterCount = 0
+            map.panel = mapIcon
+        end
+    end
+end
+
+function PANEL:UpdateRowPositions()
+    local count = #self.maps
+    local margin = 2
+    local iconWidth, iconHeight = maxIconSize( self:GetWide(), self:GetTall(), count, 1 )
+    iconWidth = iconWidth - margin * 2
+
+    local maxItemsPerRow = math.floor( self:GetWide() / iconWidth )
+    for i, row in ipairs( self.rows ) do
+        local rowNumber = i
+        local itemsLeft = count - (rowNumber - 1) * maxItemsPerRow
+        local itemsInRow = math.min( maxItemsPerRow, itemsLeft )
+        local rowWidth = (iconWidth + margin * 2) * itemsInRow
+
+        local extraSpace = math.max( 0, self:GetWide() - rowWidth )
+
+        row:SetPos( extraSpace / 2, iconHeight * (rowNumber - 1) )
+    end
+end
+
+function PANEL:OnMapClicked( _, _ )
+    -- implement
+end
+
+function PANEL:GetTotalRowWidth()
+    if #self.rows == 0 then return 0 end
+    return self.rows[1]:GetWide()
+end
+
+function PANEL:GetTotalRowHeight()
+    if #self.rows == 0 then return 0 end
+    return #self.rows * self.rows[1]:GetTall()
+end
+
+function PANEL:Flash( id )
```

Comment:
Could you double check the delays and actions of the last three?

Looks like you have two `true`s in a row and a longer delay on the last one (likely intentional?). Also the sound would be missing from the last `true`

Reply:
The delays were copied from the original mapvote, im not really sure what to do with this, readding the sound to the last one sounds look a good idea though

Reply:
Ah I missed that. I'm sure it's fine